sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/options/DecryptOptions.java

```diff
@@ -0,0 +1,148 @@
+    byte[] authenticationTag;
+
+    /**
+     * Factory method to create an instance of {@link DecryptOptions} with the given parameters.
+     *
+     * @param algorithm The algorithm to be used for decryption.
+     * @param ciphertext The content to be decrypted.
+     * @return The {@link DecryptOptions}.
+     */
+    public static DecryptOptions createOptions(EncryptionAlgorithm algorithm, byte[] ciphertext) {
+        return new DecryptOptions(algorithm, ciphertext);
+    }
+
+    /**
+     * Factory method to create an instance of {@link AesCbcDecryptOptions} with the given parameters.
+     *
+     * @param algorithm The algorithm to be used for decryption.
+     * @param ciphertext The content to be decrypted.
+     * @return The {@link AesCbcDecryptOptions}.
+     */
+    public static AesCbcDecryptOptions createAesCbcOptions(EncryptionAlgorithm algorithm, byte[] ciphertext) {
+        return new AesCbcDecryptOptions(algorithm, ciphertext);
+    }
+
+    /**
+     * Factory method to create an instance of {@link AesGcmDecryptOptions} with the given parameters.
+     *
+     * @param algorithm The algorithm to be used for decryption.
+     * @param ciphertext The content to be decrypted.
+     * @param iv Initialization vector for the decryption operation.
+     * @return The {@link AesGcmDecryptOptions}.
+     */
+    public static AesGcmDecryptOptions createAesGcmOptions(EncryptionAlgorithm algorithm, byte[] ciphertext, byte[] iv) {
+        return new AesGcmDecryptOptions(algorithm, ciphertext, iv);
+    }
+
+    /**
+     * Creates an instance of {@link DecryptOptions} with the given parameters.
+     *
+     * @param algorithm The algorithm to be used for decryption.
+     * @param cipherText The content to be decrypted.
+     */
+    DecryptOptions(EncryptionAlgorithm algorithm, byte[] cipherText) {
+        Objects.requireNonNull(algorithm, "'algorithm cannot be null'");
+        Objects.requireNonNull(cipherText, "'ciphertext' cannot be null");
+
+        this.algorithm = algorithm;
+        this.cipherText = new byte[cipherText.length];
+        System.arraycopy(cipherText, 0, this.cipherText, 0, cipherText.length);
+    }
+
+    /**
+     * The algorithm to be used for encryption.
+     *
+     * @return The algorithm to be used for encryption.
+     */
+    public EncryptionAlgorithm getAlgorithm() {
+        return algorithm;
+    }
+
+    /**
+     * Get the content to be encrypted.
+     *
+     * @return The content to be encrypted.
+     */
+    public byte[] getCipherText() {
+        if (cipherText == null) {
+            return null;
+        } else {
+            return cipherText.clone();
+        }
+    }
+
+    /**
+     * Get the initialization vector to be used in the decryption operation using a symmetric algorithm.
+     *
+     * @return The initialization vector.
+     */
+    public byte[] getIv() {
+        if (iv == null) {
+            return null;
+        } else {
+            return iv.clone();
```

Comment:
You cloned to set this. Do you need to clone again?

Reply:
Since array contents are not immutable we do not want to provide access to the original array.